php/libraries/Image.class.inc

```diff
@@ -0,0 +1,256 @@
+<?php declare(strict_types=1);
+/**
+ * File describe the Image class
+ *
+ * PHP Version 7
+ *
+ *  @category Imaging
+ *  @package  Main
+ *  @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ *  @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ *  @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS;
+
+/**
+ * This class defines a Image to allows one to type hint function's parameters
+ *
+ *  @category Imaging
+ *  @package  Main
+ *  @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ *  @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ *  @link     https://www.github.com/aces/Loris/
+ */
+class Image
+{
+    private $_fileid;
+
+    private $_filename;
+
+    private $_filelocation;
+
+    /**
+     * Constructor
+     *
+     * @param ?int $fileid This File unique id
+     */
+    public function __construct(?int $fileid)
+    {
+        if ($fileid !== null) {
+            $dbrow = \NDB_Factory::singleton()
+                ->database()
+                ->pselectRow(
+                    '
+                     SELECT
+                       f.File as filelocation,
+                       f.OutputType as outputtype,
+                       mst.Scan_type as acquisitionprotocol,
+                       FileType as filetype
+                     FROM
+                       files f 
+                     JOIN mri_scan_type mst
+                       ON (mst.ID=f.AcquisitionProtocolID)
+                     WHERE f.FileID = :v_fileid
+                    ',
+                    array('v_fileid' => $fileid)
+                );
+            if (empty($dbrow)) {
+                throw new \NotFound('There is no file with that id');
+            }
+
+            $this->_fileid       = $fileid;
+            $this->_filename     = basename($dbrow['filelocation']);
+            $this->_filelocation = $dbrow['filelocation'];
+            $this->_outputtype   = $dbrow['outputtype'];
+            $this->_acquisitionprotocol = $dbrow['acquisitionprotocol'];
+            $this->_filetype            = $dbrow['filetype'];
+        }
+    }
+
+    /**
+     * Get file headers from paramter file
+     *
+     * @param string $headername $the parameter name to reteive
```

Comment:
... $headername **the** parameter name to **retrieve** . ($ sign before `the` to remove and add a `r` in `retrieve`)